src/peergos/shared/io/ipfs/multihash/Multihash.java

```diff
@@ -69,29 +75,46 @@ public int compareTo(Multihash that) {
     }
 
     public byte[] toBytes() {
-        byte[] res = new byte[hash.length+2];
-        res[0] = (byte)type.index;
-        res[1] = (byte)hash.length;
-        System.arraycopy(hash, 0, res, 2, hash.length);
-        return res;
+        try {
+            ByteArrayOutputStream res = new ByteArrayOutputStream();
+            putUvarint(res, type.index);
+            putUvarint(res, hash.length);
+            res.write(hash);
+            return res.toByteArray();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
I think i answered my own question about exception thrown from child class Cid class before... It ain't going to happen.

Reply:
yes